### PR TITLE
api-reference: Add `exp_show` and `scm`.

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc/pull/9356

### CODE REVIEW

File: content/docs/sidebar.json [hunk 1]
```diff
@@ -566,6 +566,10 @@
       {
         "slug": "read",
         "label": "read()"
+      },
+      {
+        "slug": "scm",
+        "label": "scm"
```
Comment (dberenbaum) [inline content/docs/sidebar.json:572]: Looks like you forgot `exp_show` here?

File: content/docs/api-reference/exp_show.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+the Git `HEAD`.
+
+## Parameters
+
+- `repo` - specifies the location of the DVC project. It can be a URL or a file
+  system path. Both HTTP and SSH protocols are supported for online Git repos
+  (e.g. `[user@]server:project.git`). _Default_: The current project (found by
+  walking up from the current working directory tree).
+
+- `revs` - List of Git commit (any
```
Comment (dberenbaum) [inline content/docs/api-reference/exp_show.md:64]:
```suggestion
- `revs` - List of Git commits (any
```

File: content/docs/api-reference/exp_show.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  [revision](https://git-scm.com/docs/revisions) such as a branch or tag name, a
+  commit hash or an [experiment](/doc/command-reference/exp) name). If `repo` is
+  not a Git repo, this option is ignored. _Default_: `None` (current working
```
Comment (dberenbaum) [inline content/docs/api-reference/exp_show.md:67]: When would `repo` not be a Git repo?

File: content/docs/api-reference/exp_show.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+```py
+import dvc.api
+import pandas as pd
+
+df = pd.DataFrame(dvc.api.exp_show())
+```
+
+You can manipulate the Pandas DataFrame to use a subset of the columns and clean
+up the rows:
```
Comment (dberenbaum) [inline content/docs/api-reference/exp_show.md:86]: I hope people can find uses for this on their own, but I wonder if cleaning up the table is the best example. You can do a lot of that from the CLI also. WDYT about an example like creating a new column with the difference between training and eval metrics? That's the kind of thing I would be likely to use it for. 

Not a blocker, just an idea to consider.

**Review by dberenbaum — CHANGES_REQUESTED** [review]
Thanks @daavoo! Looks like the sidebar is broken -- not sure if that's what's causing the link errors.

**Comment by omesser** [comment]
released in [DVC 2.56.0](https://github.com/iterative/dvc/releases/tag/2.56.0)

File: content/docs/api-reference/exp_show.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+```py
+columns = ["Experiment", "dice_multi", "train.arch"]
+
+df = pd.DataFrame(dvc.api.exp_show(), columns=columns)
+
+df["Experiment"].replace("", None, inplace=True)
+df.dropna(inplace=True)
+df.reset_index(drop=True, inplace=True)
+df.to_markdown()
```
Comment (daavoo) [inline content/docs/api-reference/exp_show.md:95]:
```suggestion
df = pd.DataFrame(dvc.api.exp_show(), columns=columns)

df.dropna(inplace=True)
df.reset_index(drop=True, inplace=True)

df.to_markdown()
```

File: content/docs/api-reference/scm.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+## Usage:
+
+```py
+import dvc.api
+
+branches = dvc.api.scm.all_branches()
+commits = dvc.api.scm.all_commits()
+tags = dvc.api.scm.all_tags()
```
Comment (dberenbaum) [inline content/docs/api-reference/scm.md:26]: Can we show example output here or in the example below?

**Review by dberenbaum — APPROVED** [review]
One minor suggestion. LGTM, thanks!

**Pull request merged**